src/main/java/de/tum/cit/aet/artemis/programming/service/localvc/ArtemisGitServletService.java

```diff
@@ -53,14 +55,19 @@ public void init() {
 
         // Add filters that every request to the JGit Servlet goes through, one for each fetch request, and one for each push request.
         this.addUploadPackFilter(new LocalVCFetchFilter(localVCServletService));
-        this.addReceivePackFilter(new LocalVCPushFilter(localVCServletService));
+
+        // TODO figure out a way to keep state for an HTTP session
+        this.addReceivePackFilter(new LocalVCPushFilter(localVCServletService, (v) -> {
+            this.c = v + 1;
+            return c;
+        }));
 
         this.setReceivePackFactory((request, repository) -> {
             ReceivePack receivePack = new ReceivePack(repository);
             // Add a hook that prevents illegal actions on push (delete branch, rename branch, force push).
-            receivePack.setPreReceiveHook(new LocalVCPrePushHook(localVCServletService, (User) request.getAttribute("user")));
+            receivePack.setPreReceiveHook(new LocalVCPrePushHook(localVCServletService, (User) request.getSession().getAttribute("user")));
             // Add a hook that triggers the creation of a new submission after the push went through successfully.
-            receivePack.setPostReceiveHook(new LocalVCPostPushHook(localVCServletService));
+            receivePack.setPostReceiveHook(new LocalVCPostPushHook(localVCServletService, request));
```

Comment:
The issue here is that actually neither the request nor the session are new every step. The previously existing (User) request.getAttribute("user"))); actually never worked, as it's just always null